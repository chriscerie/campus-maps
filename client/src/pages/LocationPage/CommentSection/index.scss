.comment-section-container {
  margin-top: 40px;
  padding: 0;
  margin: 0;
}

.comment-form-title {
  font-size: 22px;
}

.comment-form-textarea {
  width: 100%;
  height: 80px;
  margin-bottom: 8px;
  margin-top: 8px;
  border: 1px solid rgb(107, 114, 112);
}
