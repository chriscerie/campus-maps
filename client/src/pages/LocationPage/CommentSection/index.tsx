import { useState, useEffect } from 'react';
import { getComments as getCommentsApi } from './api';
import { CommentType } from './Comment';
import Comment from './Comment';
import CommentForm from './CommentForm';

const CommentSection = () => {
  const [backendComments, setBackendComments] = useState([]);
  const rootComments: Array<CommentType> = backendComments.filter(
    (backendComment: CommentType) => backendComment.parentId === null
  ); //root comments are comments with no parent (is not a reply)
  const getReplies = (commentId: string) => {
    return backendComments
      .filter(
        (backendComment: CommentType) => backendComment.parentId === commentId
      )
      .sort(
        (
          a: CommentType,
          b: CommentType //sort replies by most recent -> oldest
        ) => new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime()
      );
  };
  const addComment = (text: string, parentId: string) => {
    console.log('addComment', text, parentId);
  };

  useEffect(() => {
    getCommentsApi().then((data: any) => {
      //returns array of comments from backend
      setBackendComments(data);
    });
  }, []);

  return (
    <div className="comments">
      <h3 className="comments-title">Comments</h3>
      <div className="comment-form-title">Write comment</div>
      <div className="comments-container">
        {rootComments.map((rootComment: CommentType) => (
          <Comment
            comm={rootComment}
            replies={getReplies(rootComment.id)}
            key={rootComment.id}
          /> //key, comm, replies
        ))}
      </div>
    </div>
  );
};

export default CommentSection;
