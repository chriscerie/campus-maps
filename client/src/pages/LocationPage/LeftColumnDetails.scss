#location-page-left-column {
  display: table-cell;
  width: calc(100% * 2 / 3);
  vertical-align: top;
  box-sizing: border-box;
  padding-right: 1em;
}

#location-page-left-buttons-container {
  display: flex;
  flex-wrap: wrap;
}

.location-page-left-button {
  display: flex;
  box-sizing: border-box;
  margin-right: 0.7em;
  margin-bottom: 0.7em;
}
