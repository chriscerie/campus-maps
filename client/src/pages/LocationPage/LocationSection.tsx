import { ReactNode } from 'react';
import './LocationSection.scss';

function TitleDivider({
  title,
  children,
}: {
  title: string;
  children?: ReactNode;
}) {
  return (
    <section id="location-section">
      <div id="location-section-title-container">
        <h2 id="location-section-title">{title}</h2>
      </div>
      {children}
    </section>
  );
}

export default TitleDivider;
