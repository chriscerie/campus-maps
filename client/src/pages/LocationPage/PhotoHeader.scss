.location-page-container {
  display: block;
  line-height: 2em;
}

.photo-header-container {
  position: relative;
  height: 25em;
  overflow: hidden;
  background: linear-gradient(transparent, black 110%);
}

.photo-header-content-container {
  color: #fff;
  box-sizing: border-box !important;
  align-items: flex-end;
  display: flex !important;
  position: absolute;
  padding: 2.5em 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  text-align: left;
}

.photo-header-content {
  display: inline-block;
  flex: 1 auto;
}

.photo-header-heading {
  font-size: 3em;
  font-weight: 700;
  line-height: 1.2em;
  letter-spacing: -0.4px;
}

.photo-header-subheading {
  display: inline;
  font-size: 1.4em;
  font-weight: 400;
  line-height: 1.2em;
  letter-spacing: -0.4px;
  margin-right: 0.5em;
  color: #fff;
}

.photo-header-edit-button {
  @extend .photo-header-subheading;
  text-decoration: none;
  font-size: 0.9em;
  padding: 0.35em 0.7em;
  transition: background-color 0.8s ease;
  background-color: rgba(255, 255, 255, 0.25);
  border-radius: 0.3em;
  font-weight: 600;
}

.photo-header-edit-button:hover {
  transition: background-color 0.8s ease;
  background-color: rgba(255, 255, 255, 0.4);
}

#see-more-button {
  flex: 0 auto;
  border-radius: 0.3em;
  margin-top: -0.2em;
  border: 1px solid #ddd;
  padding: 0.5em 1.7em;
  text-decoration: none;
  font-weight: 700;
  transition: background-color 0.4s ease, color 0.4s ease;
  background-color: transparent;
  color: #fff;
}

#see-more-button:hover {
  transition: background-color 0.4s ease;
  background-color: rgba(255, 255, 255, 0.3);
}

#header-image {
  z-index: -1;
}
