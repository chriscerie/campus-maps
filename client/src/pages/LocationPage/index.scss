.location-page-container {
  display: block;
  line-height: 2em;
}

#location-page-container-inner {
  display: table;
  margin-top: 30px;
}

.images-list-container {
  height: 100%;
}

.align-text-left {
  text-align: left;
}

.align-text-right {
  text-align: right;
}

.grid-item {
  padding: 16px !important;
  padding-bottom: 0px;
}

.building-info {
  padding-bottom: 0px !important;
}

.building-title-header {
  padding-top: 40px;
  padding-bottom: 0px;
  margin-bottom: 0px;
}

#header-image {
  z-index: -1;
  object-fit: contain;
  height: 100%;
  width: 100%;
  vertical-align: middle;
}

#header-image-placeholder {
  @extend #header-image;
  content: url('../../images/ucsb.jpg');
}
