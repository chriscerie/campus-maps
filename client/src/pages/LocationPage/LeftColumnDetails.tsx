import { Link } from 'react-router-dom';
import ReviewSection from './ReviewSection';
import LocationSection from './LocationSection';
import './LeftColumnDetails.scss';
import { Button } from '@mui/material';
import ChatOutlinedIcon from '@mui/icons-material/ChatOutlined';
import CameraAltOutlinedIcon from '@mui/icons-material/CameraAltOutlined';
import ScreenShareOutlinedIcon from '@mui/icons-material/ScreenShareOutlined';
import BookmarkBorderOutlinedIcon from '@mui/icons-material/BookmarkBorderOutlined';
import type { LocationType } from '../../types/LocationType';

function LeftColumnDetails({
  locationInfo,
  id,
}: {
  locationInfo: LocationType | null;
  id: string;
}) {
  return (
    <div id="location-page-left-column">
      <div id="location-page-left-buttons-container">
        <div className="location-page-left-button">
          <Link
            to={`/write-review/${locationInfo && locationInfo.id}`}
            style={{ textDecoration: 'none' }}
          >
            <Button
              variant="contained"
              disableElevation
              startIcon={<ChatOutlinedIcon />}
              sx={{ textTransform: 'none', fontSize: '1em' }}
            >
              Write a Review
            </Button>
          </Link>
        </div>
        <div className="location-page-left-button">
          <Button
            variant="outlined"
            color="inherit"
            startIcon={<CameraAltOutlinedIcon />}
            sx={{ textTransform: 'none', fontSize: '1em' }}
          >
            Add Photo
          </Button>
        </div>
        <div className="location-page-left-button">
          <Button
            variant="outlined"
            color="inherit"
            startIcon={<ScreenShareOutlinedIcon />}
            sx={{ textTransform: 'none', fontSize: '1em' }}
          >
            Share
          </Button>
        </div>
        <div className="location-page-left-button">
          <Button
            variant="outlined"
            color="inherit"
            startIcon={<BookmarkBorderOutlinedIcon />}
            sx={{ textTransform: 'none', fontSize: '1em' }}
          >
            Save
          </Button>
        </div>
      </div>

      <LocationSection title="Description">
        <div className="building-info align-text-left">
          <p style={{ margin: 0 }}>
            {locationInfo && locationInfo.description}
          </p>
        </div>
      </LocationSection>

      <LocationSection title="Classrooms">
        <div
          className="align-text-left"
          id="location-page-left-buttons-container"
        >
          {locationInfo &&
            locationInfo.rooms.map((rooms) => (
              <Link
                to={`/classrooms/${rooms && rooms.room_id}`}
                style={{ textDecoration: 'none' }}
              >
                <Button
                  variant="outlined"
                  color="inherit"
                  sx={{ textTransform: 'none', fontSize: '1em' }}
                >
                  {rooms.room_name}
                </Button>
              </Link>
            ))}
        </div>
      </LocationSection>

      <LocationSection title="Reviews">
        <div className="align-text-left">
          <ReviewSection id={id} />
        </div>
      </LocationSection>
    </div>
  );
}

export default LeftColumnDetails;
