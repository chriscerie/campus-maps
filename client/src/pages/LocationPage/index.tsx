/* eslint-disable eqeqeq */
import { Container } from '@mui/material';
import PhotoHeader from './PhotoHeader';
import { useSelector } from 'react-redux';
import { RootState } from '../../reducers';
import { useParams } from 'react-router-dom';
import './index.scss';
import { useState, useEffect } from 'react';
import LeftColumnDetails from './LeftColumnDetails';
import RightColumnDetails from './RightColumnDetails';
import Footer from '../../components/footer';

function LocationPage() {
  const { id } = useParams<{ id: string }>();
  const [selected, setSelected] =
    useState<null | mapboxgl.MapboxGeoJSONFeature>(null);
  const mapInstance = useSelector((state: RootState) => state.mapInstance.map);

  // Set point
  useEffect(() => {
    if (mapInstance) {
      const setSelectedPoint = () => {
        const points = mapInstance.querySourceFeatures('composite', {
          sourceLayer: 'poi_label',
        });

        setSelected(points.filter((point) => point.id == id)[0]);
      };

      if (mapInstance.isStyleLoaded()) {
        setSelectedPoint();
      } else {
        mapInstance.once('loaded', () => {
          setSelectedPoint();
        });
      }
    }
  }, [id, mapInstance]);

  console.log(selected);

  return (
    <div className="location-page-container">
      <PhotoHeader selected={selected} />

      <Container id="location-page-container-inner">
        <LeftColumnDetails />
        <RightColumnDetails />
      </Container>
      <Footer/>
    </div>
  );
}

export default LocationPage;
