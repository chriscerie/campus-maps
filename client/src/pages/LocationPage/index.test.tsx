import ShallowRenderer from 'react-test-renderer/shallow';
import LocationPage from './index';

describe('LocationPage', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(<LocationPage />);
    expect(tree).toMatchSnapshot();
  });
});
