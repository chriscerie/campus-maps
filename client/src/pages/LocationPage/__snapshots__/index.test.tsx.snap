// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPage should render successfully 1`] = `
<div
  className="location-page-container"
>
  <PhotoHeader />
  <ForwardRef(Container)>
    <div
      className="building-info align-text-left"
    >
      <p>
        <strong>
          Building code:
        </strong>
         PHELP 3526
        <br />
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed accumsan, purus at eleifend ornare, libero metus tincidunt velit, quis tincidunt nisl eros non tellus. Nunc ullamcorper porttitor risus, eu luctus justo auctor sed. Vestibulum ornare eros est, in tempus massa congue ut. Duis eu fringilla nulla, nec mattis ex. Quisque viverra justo a luctus feugiat. Aliquam erat volutpat. Nullam varius odio ex, non aliquet eros tristique ullamcorper. Donec dignissim nisl dui, in sollicitudin magna faucibus eu. Duis porta ornare erat, ut lacinia lacus volutpat id. Suspendisse potenti. Suspendisse et iaculis velit.
      </p>
    </div>
    <div
      className="align-text-left"
    >
      <h1>
        Comments
      </h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed accumsan, purus at eleifend ornare, libero metus tincidunt velit, quis tincidunt nisl eros non tellus. Nunc ullamcorper porttitor risus, eu luctus justo auctor sed. Vestibulum ornare eros est, in tempus massa congue ut. Duis eu fringilla nulla, nec mattis ex. Quisque viverra justo a luctus feugiat. Aliquam erat volutpat. Nullam varius odio ex, non aliquet eros tristique ullamcorper. Donec dignissim nisl dui, in sollicitudin magna faucibus eu. Duis porta ornare erat, ut lacinia lacus volutpat id. Suspendisse potenti. Suspendisse et iaculis velit.
      </p>
    </div>
  </ForwardRef(Container)>
</div>
`;
